__tests__/useSelectedItems.test.tsx

```diff
@@ -0,0 +1,155 @@
+      itemIdentifierKey: "id",
+    }));
+
+    expect(result.current.items).toEqual([]);
+  });
+
+  it("should toggle a single item", () => {
+    const { result } = renderHook(() => useSelectedItems<ExampleItem, ExampleItem["text"]>({
+      itemIdentifierKey: "id",
+      initialItems: initialExampleItems,
+    }));
+
+    act(() => {
+      result.current.toggleSingleItem(1);
+    });
+
+    expect(result.current.selectedItems).toContainEqual(
+      expect.objectContaining(initialExampleItems[0]),
+    );
+
+    act(() => {
+      result.current.toggleSingleItem(2);
+    });
+
+    expect(result.current.selectedItems).toContainEqual(
+      expect.objectContaining(initialExampleItems[0]),
+    );
+    expect(result.current.selectedItems).toContainEqual(
+      expect.objectContaining(initialExampleItems[1]),
+    );
+  });
+
+  it("should toggle a single item", () => {
+    const { result } = renderHook(() => useSelectedItems<ExampleItem, ExampleItem["text"]>({
+      itemIdentifierKey: "id",
+      initialItems: initialExampleItems,
+    }));
+
+    act(() => {
+      result.current.toggleSingleItem(1);
+    });
+
+    expect(result.current.selectedItems).toContainEqual(
+      expect.objectContaining(initialExampleItems[0]),
+    );
+
+    act(() => {
+      result.current.toggleSingleItem(2);
+    });
+
+    expect(result.current.selectedItems).toContainEqual(
+      expect.objectContaining(initialExampleItems[0]),
+    );
+    expect(result.current.selectedItems).toContainEqual(
+      expect.objectContaining(initialExampleItems[1]),
+    );
+  });
+
+  it("should toggle all items", () => {
+    const { result } = renderHook(() => useSelectedItems<ExampleItem, ExampleItem["text"]>({
+      itemIdentifierKey: "id",
+      initialItems: initialExampleItems,
+    }));
+
+    expect(result.current.selectedItems).not.toContainEqual(
+      expect.objectContaining(initialExampleItems[0]),
+    );
+
+    expect(result.current.selectedItems).not.toContainEqual(
+      expect.objectContaining(initialExampleItems[1]),
+    );
+
+    expect(result.current.selectedItems).not.toContainEqual(
+      expect.objectContaining(initialExampleItems[2]),
+    );
+
+    act(() => {
+      result.current.toggleAllItems();
+    });
+
+    expect(result.current.selectedItems).toContainEqual(
+      expect.objectContaining(initialExampleItems[0]),
+    );
+
+    expect(result.current.selectedItems).toContainEqual(
+      expect.objectContaining(initialExampleItems[1]),
+    );
+
+    expect(result.current.selectedItems).toContainEqual(
+      expect.objectContaining(initialExampleItems[2]),
+    );
+  });
+
+  it("shouldn't render if itemIdentifierKey is wrong", () => {
+    const consoleSpy = jest.spyOn(console, "error").mockImplementation();
+
+    expect(() => {
+      renderHook(() => useSelectedItems<ExampleItem, ExampleItem["text"]>({
+        itemIdentifierKey: "wrong-id",
+        initialItems: initialExampleItems,
+      }));
+    }).toThrowError();
```

Comment:
This seems to be working with the current version of `@testing-library/react-hooks` 

The hook result has an error property so it's easier to assert if an error happened or not.

```
  expect(result.error).toEqual(Error("Please, make sure to provide a valid identifier key to all items"));
```

I'll also suggest to create a constant for the error message and export it, so it can be reused in the hook module & other files.

Reply:
The error constant can be declared inside `constants.ts` and I would suggest something like that:

```
const ERRORS = {
 INVALID_UNIQUE_IDENTIFIER: "Please, make sure to provide a valid identifier key to all items"
}
```

But feel free to use your judgment here